modules/sdk-coin-algo/src/algo.ts

```diff
@@ -542,6 +606,172 @@ export class Algo extends BaseCoin {
+    }
+
+    const factory = new AlgoLib.TransactionBuilderFactory(coins.get('algo'));
+    const txBuilder = factory.getTransferBuilder();
+
+    let userPrv: string | undefined;
+    let backupPrv: string | undefined;
+    if (!isUnsignedSweep) {
+      if (!params.bitgoKey) {
+        throw new Error('bitgo public key from the keyCard is required for non-bitgo recovery');
+      }
+      try {
+        userPrv = this.bitgo.decrypt({ input: params.userKey, password: params.walletPassphrase });
+        backupPrv = this.bitgo.decrypt({ input: params.backupKey, password: params.walletPassphrase });
+        const userKeyAddress = Utils.privateKeyToAlgoAddress(userPrv);
+        const backupKeyAddress = Utils.privateKeyToAlgoAddress(backupPrv);
+        txBuilder.numberOfRequiredSigners(2).setSigners([userKeyAddress, backupKeyAddress, params.bitgoKey]);
+      } catch (e) {
+        throw new Error(
+          'unable to decrypt userKey or backupKey with the walletPassphrase provided, got error: ' + e.message
+        );
+      }
+    }
+
+    // TODO: Remove the ! after making nodeParams mandatory
+    const token =
+      this.bitgo.getEnv() === 'prod'
+        ? params.nodeParams!.token
+        : '2810c2d168e8417c5f111d38d68327b8cfe2d0ddc02986490c22f8ddf4128bcd';
+    const baseServer = this.bitgo.getEnv() === 'prod' ? params.nodeParams?.baseServer : 'http://localhost/';
+    const port = this.bitgo.getEnv() === 'prod' ? params.nodeParams?.port : 8443;
+
+    const client = new algosdk.Algodv2(token, baseServer, port);
+    const nativeBalance = await this.getAccountBalance(params.rootAddress, client);
+
+    if (new BigNumber(nativeBalance).isZero() || new BigNumber(nativeBalance).isLessThanOrEqualTo(params.fee)) {
+      throw new Error('Insufficient balance to recover, got balance: ' + nativeBalance + ' fee: ' + params.fee);
+    }
+
+    // Algorand accounts require a min. balance of 1 ALGO
+    const MIN_MICROALGOS_BALANCE = 100000;
+    const spendableAmount = new BigNumber(nativeBalance).minus(params.fee).minus(MIN_MICROALGOS_BALANCE).toNumber();
+
+    const currentRound = await client
+      .status()
+      .do()
+      .then((status) => status['last-round']);
+    const LAST_ROUND_BUFFER = 200;
+
+    const firstRound = params.firstRound ?? currentRound;
+    const lastRound = params.lastRound ?? currentRound + LAST_ROUND_BUFFER;
+
+    txBuilder
+      .fee({ fee: params.fee.toString() })
+      .isFlatFee(params.isFlatFee)
+      .sender({
+        address: params.rootAddress,
+      })
+      .to({
+        address: params.recoveryDestination,
+      })
+      .amount(spendableAmount)
+      .genesisId(genesisId)
+      .genesisHash(genesisHash)
+      .firstRound(firstRound)
+      .lastRound(lastRound);
+
+    if (params.genesisId) {
+      txBuilder.genesisId(params.genesisId);
+    }
+    if (params.lease) {
+      txBuilder.lease(params.lease);
+    }
```

Comment:
The lease is a Uint8Array, also probably not needed at all unless you want to send tx in an specific order
```suggestion
```